crates/fuel-core/src/graphql_api/worker_service.rs

```diff
@@ -429,27 +443,65 @@ where
         _: Self::TaskParams,
     ) -> anyhow::Result<Self::Task> {
         {
-            let db_tx = self.database.transaction();
+            let db_tx = self.off_chain_database.transaction();
             let total_tx_count = db_tx.get_tx_count().unwrap_or_default();
             graphql_metrics().total_txs_count.set(total_tx_count as i64);
         }
 
-        // TODO: It is possible that the node was shut down before we processed all imported blocks.
-        //  It could lead to some missed blocks and the database's inconsistent state.
-        //  Because the result of block execution is not stored on the chain, it is impossible
-        //  to actualize the database without executing the block at the previous state
-        //  of the blockchain. When `AtomicView<Storage>::view_at` is implemented, we can
-        //  process all missed blocks and actualize the database here.
-        //  https://github.com/FuelLabs/fuel-core/issues/1584
-        Ok(self)
+        let InitializeTask {
+            chain_id,
+            block_importer,
+            tx_pool,
+            on_chain_database,
+            off_chain_database,
+            import_result_provider,
+            continue_on_error,
+        } = self;
+
+        let mut task = Task {
+            tx_pool,
+            block_importer,
+            database: off_chain_database,
+            chain_id,
+            continue_on_error,
+        };
+
+        while let Some(Some(block)) = task.block_importer.next().now_or_never() {
```

Comment:
could you add comments explaining what this is doing?

Reply:
Or just move it to a helper function with a self-descriptive name.